benchmark_runner/workloads/bootstorm_vm.py

```diff
@@ -25,36 +23,66 @@ def __init__(self):
+        self.bootstorm_start_time = {}
 
-    def __create_vm_scale(self, vm_num: str):
+    def __vm_start_scale(self):
         """
-        This method creates vms in parallel
+        This method starts VMs
+        @return:
         """
-        self._oc._create_async(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num}.yaml'))
-        self._oc.wait_for_vm_status(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num}', status=VMStatus.Stopped)
+        for vm_num in range(len(self._scale_node_list) * self.__scale):
+            self.set_bootstorm_vm_start_time(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}')
+            self._virtctl.start_vm_async(vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}')
 
-    def __run_vm_scale(self, vm_num: str):
+    def __wait_for_vm_login(self, vm_num: str):
         """
-        This method runs vms in parallel
+        This method waits for VMs to enable login, in parallel
+        @return:
         """
         vm_name = f'bootstorm-vm-{self._trunc_uuid}-{vm_num}'
-        self.set_bootstorm_vm_start_time(vm_name=vm_name)
-        self._virtctl.start_vm_sync(vm_name=vm_name)
+        self._virtctl.wait_for_vm_status(vm_name=vm_name, status=VMStatus.Running)
         self.__data_dict = self.get_bootstorm_vm_elapse_time(vm_name=vm_name)
         self.__status = 'complete' if self.__data_dict else 'failed'
+        # upload to elasticsearch
         if self._es_host:
             self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status=self.__status, result=self.__data_dict)
             # verify that data upload to elastic search according to unique uuid
             self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
 
-    def __delete_vm_scale(self, vm_num: str):
+    def __create_vm_scale(self):
         """
-        This method deletes vms in parallel
+        This method creates VMs
         """
-        vm_name = f'bootstorm-vm-{self._trunc_uuid}-{vm_num}'
-        self._oc.delete_vm_sync(
-            yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num}.yaml'),
-            vm_name=vm_name)
+        # create vms
+        for vm_num in range(len(self._scale_node_list) * self.__scale):
+            self._oc.create_vm_sync(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num+1}.yaml'),
+                                    vm_name=f'bootstorm-vm-{self._trunc_uuid}-{vm_num+1}')
+
+    def __run_vm_scale(self):
+        """
+        This method start VMs in parallel and wait for login to be enabled
+        """
+        self.__vm_start_scale()
+        proc = []
+        for vm_num in range(1, (len(self._scale_node_list) * self.__scale) + 1):
+            p = Process(target=self.__wait_for_vm_login, args=(str(vm_num),))
+            p.start()
+            proc.append(p)
+            if vm_num % int(self._threads_limit) == 0:
+                time.sleep(self._oc.SLEEP_TIME)
```

Comment:
Is it possible to reap any processes that are complete at this point (equivalent to waitpid with WNOHANG)?

Reply:
I have redesign the scale run for vdbench and bootstorm workloads